src/editor/lib/entity.js

```diff
@@ -155,21 +155,25 @@ export function cloneEntityImpl(entity, newId = undefined) {
     { once: true }
   );
 
+  // In cloneEntityImpl
   if (newId) {
     clone.id = newId;
+  } else if (entity.id) {
+    clone.id =
+      entity.id.length === 21 ? createUniqueId() : getUniqueId(entity.id);
   } else {
-    if (entity.id) {
-      if (entity.id.length === 21) {
-        // nanoid generated id, create a new one
-        clone.id = createUniqueId();
-      } else {
-        // Get a valid unique ID for the entity
-        clone.id = getUniqueId(entity.id);
-      }
-    } else {
-      entity.id = createUniqueId();
+    clone.id = createUniqueId();
+  }
+
+  // Add recursive ID generation for children
+  function regenerateIDs(element) {
+    if (element.id) {
+      element.id = createUniqueId();
     }
+    Array.from(element.children).forEach(regenerateIDs);
   }
+  regenerateIDs(clone);
```

Comment:
We need to have children ids to be the same if you do undo entityupdate of child, undo entityclone, redo entityclone, redo entityupdate of child.